File: leetcode/python/clear-digits.py
#!/usr/bin/env python3

""" Leetcode problem 3174: Clear Digits. """

from collections import deque


class Solution:
    def clearDigits(self, s: str) -> str:
        stack = deque()

        for character in s:
            if character.isdigit():
                try:
                    stack.pop()
                except IndexError:
                    pass
            else:
                stack.append(character)
        return ''.join(stack)


def main() -> None:
    pass


if __name__ == "__main__":
    main()

File: leetcode/rust/clear-digits.rs
// Leetcode problem 3174: Clear Digits.

use std::collections::VecDeque;

struct Solution {}

impl Solution {
    pub fn clear_digits(s: String) -> String {
        let mut queue: VecDeque<char> = VecDeque::with_capacity(s.len());
        s.chars().for_each(|x| {
            if x.is_numeric() {
                queue.pop_back();
            } else {
                queue.push_back(x);
            }
        });
        queue.iter().collect()
    }
}

fn main() {}
